test/support/single_cov.rb

```diff
@@ -0,0 +1,67 @@
+class SingleCov
+  COVERAGES = []
+
+  class << self
+    def expect(file, percent)
+      COVERAGES << [file, percent]
+    end
+
+    def verify!(result)
+      COVERAGES.each do |file, expected_percent|
+        uncovered!(file) unless coverage = result[File.expand_path(file)]
+
+        line_count = coverage.count { |l| l }
```

Comment:
we want to ignore the nils ... just count would not do that

On Mon, Apr 11, 2016 at 9:35 PM, Roman Sandler <notifications@github.com>
wrote:

> In test/support/single_cov.rb
> <https://github.com/zendesk/samson/pull/805#discussion_r59319705>:
>
> > @@ -0,0 +1,67 @@
> > +class SingleCov
> > +  COVERAGES = []
> > +
> > +  class << self
> > +    def expect(file, percent)
> > +      COVERAGES << [file, percent]
> > +    end
> > +
> > +    def verify!(result)
> > +      COVERAGES.each do |file, expected_percent|
> > +        uncovered!(file) unless coverage = result[File.expand_path(file)]
> > +
> > +        line_count = coverage.count { |l| l }
>
> I guess we don't need to pass the block here? Or are some of these
> elements false?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/805/files/f77d1b4a4260626d6b493ac6ea4f4aaa0cf1fc7e#r59319705>
>